src/sagas/networkSettings.js

```diff
@@ -34,64 +35,69 @@ export function* updateNetworkSettings(action) {
     walletServiceWsUrl,
   } = action.payload || {};
 
+  const errors = {};
+
   // validates input emptyness
   if (isEmpty(action.payload)) {
-    yield put(networkSettingsUpdateFailure());
-    return;
+    errors.message = t`Custom Network Settings cannot be empty.`;
   }
 
   // validates explorerUrl
   // - required
   // - should have a valid URL
   if (isEmpty(explorerUrl) || invalidUrl(explorerUrl)) {
-    yield put(networkSettingsUpdateFailure());
-    return;
+    errors.explorerUrl = t`explorerUrl should be a valid URL.`;
   }
 
   // validates explorerServiceUrl
   // - required
   // - should have a valid URL
   if (isEmpty(explorerServiceUrl) || invalidUrl(explorerServiceUrl)) {
-    yield put(networkSettingsUpdateFailure());
-    return;
+    errors.explorerServiceUrl = t`explorerServiceUrl should be a valid URL.`;
   }
 
   // validates nodeUrl
   // - required
   // - should have a valid URl
   if (isEmpty(nodeUrl) || invalidUrl(nodeUrl)) {
-    yield put(networkSettingsUpdateFailure());
-    return;
+    errors.nodeUrl = t`nodeUrl should be a valid URL.`;
   }
 
   // validates walletServiceUrl
   // - optional
   // - should have a valid URL, if given
   if (walletServiceUrl && invalidUrl(walletServiceUrl)) {
-    yield put(networkSettingsUpdateFailure());
-    return;
+    errors.walletServiceUrl = t`walletServiceUrl should be a valid URL.`;
   }
 
   // validates walletServiceWsUrl
   // - conditionally required
   // - should have a valid URL, if walletServiceUrl is given
   if (walletServiceUrl && invalidUrl(walletServiceWsUrl)) {
-    yield put(networkSettingsUpdateFailure());
+    errors.walletServiceWsUrl = t`walletServiceWsUrl should be a valid URL.`;
+  }
+
+  // TODO: Refactor by segregating Failure from Errors
+  // - create networkSettingsUpdateErrors
+  // - implement reaction to networkSettingsUpdateFailure
+  yield put(networkSettingsUpdateErrors(errors));
+  if (Object.keys(errors).length > 0) {
     return;
   }
 
   // NOTE: Should we allow that all the URLs be equals?
   // In practice they will never be equals.
 
+  const networkSettings = yield select((state) => state.networkSettings);
   const backupUrl = {
-    nodeUrl: config.getServerUrl(),
-    explorerServiceUrl: config.getExplorerServiceBaseUrl(),
-    walletServiceUrl: config.getWalletServiceBaseUrl(),
-    walletServiceWsUrl: config.getWalletServiceBaseWsUrl(),
+    nodeUrl: networkSettings.nodeUrl,
+    explorerServiceUrl: networkSettings.explorerServiceUrl,
```

Comment:
Missing explorer URL?

Reply:
If yes, then you must add also to the rollback method

Reply:
No. The config doesn't have a set method for explorer URL, see:
![image](https://github.com/HathorNetwork/hathor-wallet-mobile/assets/5992210/21025a6f-2290-451b-a19b-3a17e9fed58f)